source/MRMesh/MRMeshProject.cpp

```diff
@@ -105,6 +105,71 @@ MeshProjectionResult findProjection( const Vector3f & pt, const MeshPart & mp, f
     return findProjectionSubtree( pt, mp, mp.mesh.getAABBTree(), upDistLimitSq, xf, loDistLimitSq, validFaces, validProjections );
 }
 
+void findTrisInBall( const MeshPart & mp, Ball ball, const FoundTriCallback& foundCallback, const FacePredicate & validFaces )
+{
+    const auto & tree = mp.mesh.getAABBTree();
+    if ( tree.nodes().empty() )
+        return;
+
+    constexpr int MaxStackSize = 32; // to avoid allocations
+    NodeId subtasks[MaxStackSize];
+    int stackSize = 0;
+
+    auto boxInBall = [&]( NodeId n )
+    {
+        return ( tree.nodes()[n].box.getBoxClosestPointTo( ball.center ) - ball.center ).lengthSq() < ball.radiusSq;
+    };
```

Comment:
is it always valid?
![image](https://github.com/user-attachments/assets/678c21aa-959c-4f97-8e7b-4ba57a3d7363)